compiler/rustc_typeck/src/check/pat.rs

```diff
@@ -649,39 +649,38 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
-    fn borrow_pat_suggestion(
-        &self,
-        err: &mut Diagnostic,
-        pat: &Pat<'_>,
-        inner: &Pat<'_>,
-        expected: Ty<'tcx>,
-    ) {
+    fn borrow_pat_suggestion(&self, err: &mut Diagnostic, pat: &Pat<'_>, inner: &Pat<'_>) {
```

Comment:
What is `pat` and `inner` in this case? Is `pat` the (direct?) parent of `inner` always?

Reply:
Yes. `pat` is `Ref(inner)`, see 
https://github.com/rust-lang/rust/blob/e9d49b2bfc4e4c327987bb9d2a65b113b10100de/compiler/rustc_typeck/src/check/pat.rs#L201-L202

https://github.com/rust-lang/rust/blob/e9d49b2bfc4e4c327987bb9d2a65b113b10100de/compiler/rustc_typeck/src/check/pat.rs#L1871

Reply:
Huh, could we simplify the redundant parameters then, if you think that would be simpler? Or if not, could you leave a comment that explains this relationship.

Reply:
What we really want is to have `pat: Pat<PatKind::Ref>` or something, but we don't have variant types so everything is messy :(

In the end I've removed `inner` (we need to match anyway, so that should be fine) and added comments about preconditions